src/t8_cmesh/t8_cmesh_readmshfile.c

```diff
@@ -776,6 +850,14 @@ t8_cmesh_from_msh_file (const char *fileprefix, int partition,
       t8_global_errorf ("Could not open file %s\n", current_file);
```

Comment:
Definitely not your fault, but while checking this i notices that a cmesh destroy is also missing in this line, causing a memory leak when the file cannot be opened.
Could you please add this as well?